### DISCUSSION TITLE
Preprocessing Dandelion

### DISCUSSION THREAD

**Discussion by nezirajt** [tuonglab/dandelion#381] [General]
Hi, I have been using dandelion for BCR analysis. I noticed that after performing pre-processing (including reassign alleles and isotypes), the total number of contigs is reduced when merging the output filtered_contig_dandelion.tsv files. Will contigs be removed during these steps? Thanks a lot!

**Comment by zktuong** [comment]
Hi @nezirajt, are you working on mouse data? Can you provide me with some more descriptions like what does the logs show?

**Comment by nezirajt** [comment]
Thanks for your quick reply! I am working on human data, and as it seems when reassigning alleles those contigs that fail during the "creating germline" step are removed: 

      Reassigning alleles
null device 
          1 
     START> CreateGermlines
      FILE> test_y_heavy_igblast_db-pass_genotyped.tsv
GERM_TYPES> dmask
 SEQ_FIELD> sequence_alignment
   V_FIELD> v_call_genotyped
   D_FIELD> d_call
   J_FIELD> j_call
    CLONED> False

PROGRESS> 21:00:06 |####################| 100% (3,280) 0.0 min

 OUTPUT> test_y_heavy_igblast_db-pass_genotyped_germ-pass.tsv
RECORDS> 3280
   PASS> 3239
   FAIL> 41
    END> CreateGermlines

For example in this case 41. If I merge the "/dandelion/tmp/filtered_contig_igblast_db-pass.tsv" files (so before performing these steps) vs. "/dandelion/filtered_contig_dandelion.tsv" (after performing these steps), in the second case I will have less of contigs in total.

Do I have to specify the human gremline reference in this step:

ddl.pp.reassign_alleles(Sample, combined_folder="test")?

Thanks a lot!

**Reply by zktuong** [reply]
The default is human so you wouldn't have to toggle that.

The reason why the `reassign_alleles` step will drop contigs is because it does try to run create germlines internally, so if any contig's germline sequence cannot be reconstructed, they will be removed. `reassign_alleles` is an optional step though - you can skip it and go straight to `assign_isotypes`.

having the germline reconstructed is essential for downstream mutation quantification.
